src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.SymbolicLink.cs

```diff
@@ -0,0 +1,129 @@
+    [ConditionalClass(typeof(SymbolicLink_Changed_Tests), nameof(CanCreateSymbolicLinks))]
+    public class SymbolicLink_Changed_Tests : FileSystemWatcherTest
+    {
+        private string CreateSymbolicLinkToTarget(string targetPath, bool isDirectory, string linkPath = null)
+        {
+            linkPath ??= GetTestFilePath();
+            Assert.True(CreateSymLink(targetPath, linkPath, isDirectory));
+
+            return linkPath;
+        }
+
+        [Fact]
+        public void FileSystemWatcher_FileSymbolicLink_TargetsFile_Fails()
+        {
+            // Arrange
+            using var tempFile = new TempFile(GetTestFilePath());
+            string linkPath = CreateSymbolicLinkToTarget(tempFile.Path, isDirectory: false);
+
+            // Act - Assert
+            Assert.Throws<ArgumentException>(() => new FileSystemWatcher(linkPath));
+        }
+
+        // Windows 7 and 8.1 doesn't throw in this case, see https://github.com/dotnet/runtime/issues/53010.
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows10OrLater))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void FileSystemWatcher_DirectorySymbolicLink_TargetsFile_Fails()
+        {
+            // Arrange
+            using var tempFile = new TempFile(GetTestFilePath());
+            string linkPath = CreateSymbolicLinkToTarget(tempFile.Path, isDirectory: true);
+            using var watcher = new FileSystemWatcher(linkPath);
+
+            // Act - Assert
+            Assert.Throws<FileNotFoundException>(() => watcher.EnableRaisingEvents = true);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void FileSystemWatcher_DirectorySymbolicLink_TargetsSelf_Fails()
+        {
+            // Arrange
+            string linkPath = GetTestFilePath();
+            CreateSymbolicLinkToTarget(targetPath: linkPath, isDirectory: true, linkPath: linkPath);
+            using var watcher = new FileSystemWatcher(linkPath);
+
+            // Act - Assert
+            Assert.Throws<FileNotFoundException>(() => watcher.EnableRaisingEvents = true);
+        }
+
+        [Fact]
+        public void FileSystemWatcher_SymbolicLink_TargetsDirectory_Create()
+        {
+            // Arrange
+            using var tempDir = new TempDirectory(GetTestFilePath());
+            string linkPath = CreateSymbolicLinkToTarget(tempDir.Path, isDirectory: true);
+
+            using var watcher = new FileSystemWatcher(linkPath);
+            watcher.NotifyFilter = NotifyFilters.DirectoryName;
+
+            string subDirName = GetTestFileName();
+            string subDirPath = Path.Combine(tempDir.Path, subDirName);
+
+            // Act - Assert
+            ExpectEvent(watcher, WatcherChangeTypes.Created,
+                action: () => Directory.CreateDirectory(subDirPath),
+                cleanup: () => Directory.Delete(subDirPath),
+                expectedPath: Path.Combine(linkPath, subDirName));
```

Comment:
Super optional nit:

```suggestion
            string subDirPath = Path.Combine(linkPath, subDirName);

            // Act - Assert
            ExpectEvent(watcher, WatcherChangeTypes.Created,
                action: () => Directory.CreateDirectory(subDirPath),
                cleanup: () => Directory.Delete(subDirPath),
                expectedPath: subDirPath);
```

Reply:
I specifically wanted to use the "real" path to create the file system objects and expect the "link" path being reported in the watcher. I don't think it makes any difference to use the "link" path for the former.